ui/components/component-library/checkbox/checkbox.tsx

```diff
@@ -56,7 +57,7 @@ export const Checkbox: CheckboxComponent = React.forwardRef(
         alignItems={AlignItems.center}
         ref={ref}
         htmlFor={id}
-        {...props} // TODO: There is a typing issue with spreading props to the Box component. It still works but TypeScript complains.
+        {...props}
```

Comment:
```suggestion
        ...(props as Checkbox<C>)
```

Reply:
This already has an `as="label"` property, do we want to continue to override it?

Reply:
I also don't really see how this would work.

Reply:
Ah apologies. I see your point. I'll look into it further

Reply:
Apologies on the lazy ill-informed initial review again 🙇